### PR TITLE
Feature: support customizing bind-address when allow-lan is true

### PR DESCRIPTION
**Pull request by ruisiji**
fixes https://github.com/Dreamacro/clash/issues/135

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -11,6 +11,7 @@ env:
     - GO111MODULE=on
 script:
   - go test ./...
+  - make linux-amd64
```
Comment (Dreamacro) [inline .travis.yml]: Why is it built?

File: constant/general.go [hunk 2]
```diff
@@ -0,0 +1,4 @@
+package constant
```
Comment (Dreamacro) [inline constant/general.go]: Variables cannot be defined here

File: proxy/listener.go [hunk 3]
```diff
@@ -5,13 +5,15 @@ import (
...
 
+	C "github.com/Dreamacro/clash/constant"
 	"github.com/Dreamacro/clash/proxy/http"
 	"github.com/Dreamacro/clash/proxy/redir"
 	"github.com/Dreamacro/clash/proxy/socks"
 )
 
 var (
-	allowLan = false
+	allowLan    = C.AllowLan
```
Comment (Dreamacro) [inline proxy/listener.go]: use `clash/constant` value let the data flow becomes chaotic

File: proxy/listener.go [hunk 4]
```diff
@@ -97,10 +105,6 @@ func reCreateSocksUDP(port int) error {
 		socksUDPListener = nil
 	}
 
-	if portIsZero(addr) {
```
Comment (Dreamacro) [inline proxy/listener.go]: why remove this check function

File: .travis.yml (hunk 1 above)
Comment (ruisiji) [reply .travis.yml]: this can make the ci more reliable.

File: proxy/listener.go (hunk 4 above)
Comment (ruisiji) [reply proxy/listener.go]: because this check is done in the caller function, no need here.

File: constant/general.go (hunk 2 above)
Comment (ruisiji) [reply constant/general.go]: why can not use constant for these two variables? could you give me more info?

File: proxy/listener.go (hunk 3 above)
Comment (ruisiji) [reply proxy/listener.go]: could you provide more info about this?

File: .travis.yml (hunk 1 above)
Comment (Dreamacro) [reply .travis.yml]: Is there anything that `go test` cannot do and `make` can do?

File: constant/general.go (hunk 2 above)
Comment (Dreamacro) [reply constant/general.go]: `clash/constant` is used to store abstract instead of the default config value

File: proxy/listener.go (hunk 3 above)
Comment (Dreamacro) [reply proxy/listener.go]: `allowLan` in `clash/proxy` just a private variable, It doesn't really matter what value it initially has.

File: proxy/listener.go (hunk 4 above)
Comment (Dreamacro) [reply proxy/listener.go]: I don't see where to validate this value

Comment (ruisiji) [reply proxy/listener.go]: caller function: https://github.com/Dreamacro/clash/blob/288afd1308b95f61585233dd5bdc9bbebe8f96de/proxy/listener.go#L65

validation: https://github.com/Dreamacro/clash/blob/288afd1308b95f61585233dd5bdc9bbebe8f96de/proxy/listener.go#L76

call function: https://github.com/Dreamacro/clash/blob/288afd1308b95f61585233dd5bdc9bbebe8f96de/proxy/listener.go#L86

validation in call function(no need): https://github.com/Dreamacro/clash/blob/288afd1308b95f61585233dd5bdc9bbebe8f96de/proxy/listener.go#L100

**Comment by Dreamacro** [comment]
thanks for your contribution 👏

**Pull request merged**